### DISCUSSION TITLE
Feature for consideration: optional prompt character(s).

### DISCUSSION THREAD

**Discussion by brucemack** [philj404/SimpleSerialShell#17] [Ideas]
Would it be reasonable to add the ability to define a prompt string that would be displayed whenever the shell was waiting for user input?  I would typically install something like "> " or "$ ".

Why?  Two reasons:

1. It makes the shell a bit more user-friendly, since it lets the user know when the system is ready to accept input (as opposed to still crunching on the previous command).  This pattern is used by the standard UNIX shells.
2. Makes it easier to write a parser for serial automation of a device that uses the shell.  The prompt would serve as a delimiter that indicates that the output of the previous command is finished and that a new command can be sent.  This would be helpful for high-level system tests where a device is being controlled via its serial port.

I'll submit a PR for this if the idea is acceptable, but I want to check this first.

**Comment by philj404** [comment]
This seems like a straightforward, low overhead and low risk change.

For some reason I was happy with just the "OK" response.

Is it really necessary to allow customization?  The more customization, the harder it is to find the SIMPLE part.

**Reply by brucemack** [reply]
Hi Phil: I want to make sure I understand the intent of the library, because I might be missing something.  As far as I can see, the OK prompt is only displayed when someone presses \r without entering any command.  If a command runs successfully there is no OK response presented.  

The SIMPLE goal is very important (that's why the library is so good), so I want to minimize complexity.  My real goal isn't to be able to customize the prompt per-se, it's mainly to allow a predictable prompt **of some kind** to be turned on.  Would it be better to have a feature that simply enabled the "OK" prompt after any command finishes executing?  That would certainly fix my issue.

My suggestion about allowing the prompt to be set to something like "> " or "$ " was simply to make things more consistent with a typical shell experience.

**Reply by philj404** [reply]
Would it be best to just hard-code a prompt?  I like the idea of just making the prompt something obvious and visible like `"> "`.  For some reason I just never though of it -- pressing "Enter" and seeing "OK" gave me enough feedback to know the shell was working.

While supporting custom prompts isn't that much extra software effort, it... maybe sends the wrong signal?  Allowing customization complicates the interface, which in turn makes it harder to find the more important features of SimpleSerialShell.

That's a bigger topic.  I need to start a Discussion about "What is 'Simple Enough'?"  or maybe "What is 'Too Simple'?"

**Reply by brucemack** [reply]
OK, no prob.  I will send you a PR for enabling an optional, but non-configurable "> " prompt.  That will fully address my need.  You can decide whether this is making the library too complicated.  Thanks, Bruce.

**Comment by philj404** [comment]
I just moved this to a new "Discussion" section.  It's not really a problem with the existing implementation, it's an enhancement worth talking about.

**Reply by philj404** [reply]
I meant to open PR #23 to hold a sanity-check/discussion before the merge, but then things got ahead of me.  So a "`> `" command prompt is in place now, as of release 0.9.0.
If something still is not right, let me know.

**Reply by brucemack** [reply]
All good, much appreciated.
